rules/prebuilt_framework.bzl

```diff
@@ -0,0 +1,93 @@
+    """
+
+    visibility = kwargs.pop("visibility", None)
+    tags = kwargs.pop("tags", [])
+
+    legacy_target_label = "_" + name
+    apple_dynamic_framework_import(
+        name = legacy_target_label,
+        tags = tags + ["manual"],
+        **kwargs
+    )
+
+    target_patcher(
+        name = name,
+        legacy_target = legacy_target_label,
+        isDynamic = True,
+        visibility = visibility,
+        tags = tags,
+    )
+
+def apple_static_framework_import_patched(name, **kwargs):
+    """Patches an apple_static_framework_import target based on the problems reported in https://github.com/bazel-ios/rules_ios/issues/55
+
+    Args: same as the ones of apple_static_framework_import
+    """
+
+    visibility = kwargs.pop("visibility", None)
+    tags = kwargs.get("tags", [])
+
+    legacy_target_label = "_" + name
+    apple_static_framework_import(
+        name = legacy_target_label,
+        tags = tags + ["manual"],
+        **kwargs
+    )
+
+    target_patcher(
+        name = name,
+        legacy_target = legacy_target_label,
+        isDynamic = False,
+        visibility = visibility,
+        tags = tags,
+    )
+
+def _target_patcher_impl(ctx):
+    old_objc_provider = ctx.attr.legacy_target[apple_common.Objc]
+
+    # Merge providers
+    objc_provider_fields = {
+        "providers": [old_objc_provider],
+        # Adding the module_map to the headers.
+        # This "hack" is just a way to propagate the module map. swift_rules has a field in its provider
+        # called swiftc_inputs where you can pass extra dependencies to the compiler. Unfortunately
+        # the objc provider has no such field. The "hack" is to pass the module map as a header so that
+        # it makes it to the list of dependencies. Since it doesn't end in .h it gets ignored but it's
+        # still propagated as a dependency. This "hack" is based on:
+        # a) https://github.com/bazelbuild/rules_apple/blob/313eeb838497e230f01a7367ae4555aaf0cac62e/apple/internal/apple_framework_import.bzl#L98
+        # b) https://github.com/bazel-ios/rules_ios/blob/1755c694f8d5cc1ed256bba7ccf122a3fbc4addf/rules/framework.bzl#L247
+        "header": old_objc_provider.module_map,
+    }
+    new_objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
+
+    # I can not find a way to iterate on the existing providers, so instead I list here the keys to all of them
+    # (you can see the keys for the existing providers of a target by just printing the target)
+    other_provider_keys = []
+    if ctx.attr.isDynamic:
+        other_provider_keys = [AppleFrameworkImportInfo, apple_common.AppleDynamicFramework, CcInfo, OutputGroupInfo]
+    else:
+        other_provider_keys = [AppleFrameworkImportInfo, SwiftUsageInfo, CcInfo, OutputGroupInfo]
+    return [ctx.attr.legacy_target[provider_key] for provider_key in other_provider_keys] + [new_objc_provider]
+
+target_patcher = rule(
```

Comment:
`apple_framework_import_modulemap` might be a more descriptive name?

Reply:
Sure, will update to `apple_framework_import_modulemap`